onnxruntime/core/optimizer/matmul_scale_fusion.cc

```diff
@@ -256,6 +256,7 @@ Status ProcessNode(
     }
   }
```

Comment:
Site question: Is using graph_utils::FinalizeNodeFusion an option here to handle moving output edges and node removal? conv_add_fusion.cc has example usage for reference.

Reply:
I tried to use FinalizeNodeFusion initially, but it didn't support the moving of inputs from multiple nodes to the fused node. So I chose to explicitly set the fused node inputs and outputs and duplicate the node removal logic from FinalizeNodeFusion here.

Reply:
FWIW it would be good to augment FinalizeNodeFusion or add something to graph_utils that does support moving inputs from multiple nodes. Next person who copies this implementation will duplicate the logic one more time.

Reply:
That's a good idea. It's not the first time I've run into difficulties trying to use FinalizeNodeFusion(). Perhaps when I have time to understand transformer and Graph interactions better I can give it a try.